k8scrdclient/k8scrdclient.go

```diff
@@ -153,3 +164,61 @@ func (c *CRDClient) ensureStatusSubresourceCreated(ctx context.Context, crd *api
 
 	return nil
 }
+
+func crdAPIVersionEqual(a, b *apiextensionsv1beta1.CustomResourceDefinition) bool {
+	if a == nil && b == nil {
+		return true
+	}
+
+	if a == nil || b == nil {
+		return false
+	}
```

Comment:
Since those two checks are repeated in each `crd*Equal` function we can move them to `ensureUpdated` method and check that once.